src/exchanges/adapters/ZeroExV2Adapter.sol

```diff
@@ -230,4 +230,46 @@ contract ZeroExV2Adapter is ExchangeAdapter, OrderTaker {
             assetAddress_ := mload(add(_assetData, 36))
         }
     }
+
+    /// @notice Decode the parameters of a takeOrder call
+    /// @param _encodedArgs Encoded parameters passed from client side
+    /// @return orderAddresses needed addresses for an exchange to take an order
+    /// - [0] 0x Order param: makerAddress
+    /// - [1] 0x Order param: takerAddress
+    /// - [2] 0x Order param: feeRecipientAddress
+    /// - [3] 0x Order param: senderAddress
+    /// @return orderValues needed values for an exchange to take an order
+    /// - [0] 0x Order param: makerAssetAmount
+    /// - [1] 0x Order param: takerAssetAmount
+    /// - [2] 0x Order param: makerFee
+    /// - [3] 0x Order param: takerFee
+    /// - [4] 0x Order param: expirationTimeSeconds
+    /// - [5] 0x Order param: salt
+    /// - [6] Taker asset fill quantity
+    /// @return orderData Need data for an exchange to take an order
+    /// - [0] 0x Order param: makerAssetData
+    /// - [1] 0x Order param: takerAssetData
+    /// @return signature Signature of the order
+    function __decodeTakeOrderArgs(
+        bytes memory _encodedArgs
+    )
+        public
+        pure
+        returns (
+            address[4] memory orderAddresses,
+            uint256[7] memory orderValues,
+            bytes[2] memory orderData,
+            bytes memory signature
```

Comment:
For both 0x adapters, it might be nice to make `uint256 takerFillAmount` its own decoded value, as it doesn't refer to the 0x order, but is the desired fill amount for taking this order. As mentioned above, we could also have `uint256 makerFillAmount` (however named), so that we can perform extra validation, but that would be optional / not strictly-necessary.

Maybe even re-name the params `zeroExOrderAddresses`, etc, so it's clear that these params are specific to the 0x order, not the Melon order.